Appraisals

```diff
@@ -1452,7 +1452,7 @@ elsif ruby_version?('3.0') || ruby_version?('3.1')
     gem 'mongo', '>= 2.8.0', '< 2.15.0' # TODO: FIX TEST BREAKAGES ON >= 2.15 https://github.com/DataDog/dd-trace-rb/issues/1596
     gem 'mysql2', '>= 0.5.3', platform: :ruby
     gem 'pg', '>= 1.1', platform: :ruby
-    # gem 'racecar', '>= 0.3.5' # Pending release of our fix: https://github.com/appsignal/rdkafka-ruby/pull/144
+    gem 'racecar', '>= 0.3.5'
```

Comment:
1. Installing `racecar` in appraisal group should eventually ends up changes at the corresponding lock file.  I suggested this to be done after our [current progress of changing images](https://github.com/DataDog/dd-trace-rb/pull/2407). 
2. Should also enable CI with changes in `Rakefile`

Reply:
1. Merged main branch which already has 3.0.4 changes
2. Rakefile changes implemented to include all version 3.0 and above